external/fv3fit/fv3fit/reservoir/adapters.py

```diff
@@ -198,3 +206,97 @@ def load(cls, path: str) -> "HybridReservoirDatasetAdapter":
             model=model,
         )
         return adapter
+
+
+ReservoirAdapterType = Union[ReservoirDatasetAdapter, HybridReservoirDatasetAdapter]
+ReservoirModelLike = Union[ReservoirModelType, ReservoirAdapterType]
+
+
+@typing.no_type_check
+def split_multi_subdomain_model(
```

Comment:
Initially this was an attempt to contain the changes rather than add another method to the 4 model classes, but I agree the conditionals are cumbersome.  I made the change to move it into the pure reservoir and did my best not to repeat too much code.  I think this points to the need for some consolidation of the reservoir models and adapters, but I'll leave that for a future refactor.